simulation_server.py

```diff
@@ -438,14 +452,30 @@ def kill_webots(self):
                     self.webots_process.kill()
                 self.webots_process = None
 
-            # remove unused _webots images
-            available_images = os.popen(
-                "docker images --filter=reference='*_webots:*' --format '{{.Repository}}'").read().split('\n')
-            running_images = os.popen("docker ps --format '{{.Image}}'").read().split('\n')
-            unused_images = ' '.join([i for i in available_images if i not in running_images])
-            if unused_images:
-                os.system(f"docker image rm {unused_images}")
             # remove dangling images, stopped containers, build cache, volumes and networks
+            # Get list of all images
+            output = subprocess.check_output(['docker', 'images', '-a', '--format', '{{json . }}'])
+            output = output.decode()
+
+            # Split output into individual JSON objects
+            image_strings = output.strip().split('\n')
+            images = []
+            for image_string in image_strings:
+                images.append(json.loads(image_string))
+            current_time = time.time()
+
+            for image in images:
+                repository = image['Repository']
+                tag = image['Tag']
+                created_at = image['CreatedAt']
+                image_id = image['ID']
+                created_at = time.mktime(time.strptime(created_at, '%Y-%m-%d %H:%M:%S %z %Z'))
+                # Check if image is not in use by any running containers and if it was created more than 24 hours ago
+                output = subprocess.check_output(['docker', 'ps', '-q', '-f', f'ancestor={image_id}'])
+                if (output == b'' and (current_time - created_at) > 24 * 60 * 60
+                        and f'{repository}:{tag}' not in config['persistantDockerImages']):
```

Comment:
The indentation of this line looks wrong...

Reply:
I followed flake8 rules: I broke the long line [before the binary operator](https://peps.python.org/pep-0008/#should-a-line-break-before-or-after-a-binary-operator) and I added an indent for avoid [E129](https://www.flake8rules.com/rules/E129.html)